src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy

```diff
@@ -0,0 +1,186 @@
+package life.qbic.datamodel.datasets
+
+import life.qbic.datamodel.datasets.datastructure.files.DataFile
+import life.qbic.datamodel.datasets.datastructure.folders.DataFolder
+import life.qbic.datamodel.datasets.datastructure.folders.nfcore.ProcessFolder
+import life.qbic.datamodel.identifiers.SampleCodeFunctions
+
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+
+/**
+ * Class that holds information about a the output of a nf-core pipeline
+ *
+ * @since 2.5.0
+ */
+final class NfCoreExperiment {
+
+    // Fully qualified domain name of the nfcore folder structure package
+    private final static String FQDN_FOLDERS = "life.qbic.datamodel.datasets.datastructure.folders.nfcore"
+    // Fully qualified domain name of the nfcore file structure package
+    private final static String FQDN_FILES = "life.qbic.datamodel.datasets.datastructure.files.nfcore"
+
+    private final List<?> resultSet
+
+    private final static Set nfCoreFileTypes = [
+            FQDN_FILES + ".ExecutionReport",
+            FQDN_FILES + ".SampleIds",
+            FQDN_FILES + ".PipelineReport",
+            FQDN_FILES + ".SoftwareVersions",
+            FQDN_FILES + ".RunId"
+    ]
+
+    private final static Set nfCoreFolderTypes = [
+            FQDN_FOLDERS + ".PipelineInformationFolder",
+            FQDN_FOLDERS + ".QualityControlFolder",
+            FQDN_FOLDERS + ".ProcessFolder"
+    ]
+
+    private NfCoreExperiment(List<?> resultSet) {
+        this.resultSet = Objects.requireNonNull(resultSet as Object, "resultSet must not be null") as List<?>
+    }
```

Comment:
An nf core output is the result set. And this result set has properties, as described in the ER.

Reply:
Done!